studio/src/editors/merch-card-collection-editor.js

```diff
@@ -0,0 +1,862 @@
+                    });
+
+                    // Check if the reference already exists
+                    const existingReference = this.fragment.references?.find(
+                        (ref) => ref.path === fragmentData.path,
+                    );
+
+                    if (!existingReference) {
+                        // Add the new reference
+                        this.fragment.references = [
+                            ...(this.fragment.references || []),
+                            fragmentData,
+                        ];
+
+                        // Create a FragmentStore for the new reference
+                        this.fragmentReferencesMap.set(
+                            fragmentData.path,
+                            new FragmentStore(new Fragment(fragmentData)),
+                        );
+                    }
+
+                    // Remove dragover class from all elements
+                    this.#removeAllDragoverClasses();
+
+                    this.requestUpdate();
+                }
+            } catch (error) {
+                console.error('Error handling external drop:', error);
+            }
+
+            return;
+        }
+
+        // Prevent dropping cards into collections and vice versa
+        if (this.draggingFieldName !== FIELD_MODEL_MAPPING[model.path]) {
+            console.warn('Cannot drop items between different sections');
+            return;
+        }
+
+        const currentFieldName = FIELD_MODEL_MAPPING[model.path];
+
+        // Get the values from the current field
+        const field = this.fragment.fields.find(
+            (field) => field.name === currentFieldName,
+        );
+
+        if (!field?.values?.length) {
+            console.error(`Field ${currentFieldName} has no values`);
+            return;
+        }
+
+        // Get the dragged item path
+        const draggedPath = field.values[this.draggingIndex];
+        if (!draggedPath) {
+            console.error(`No item found at index ${this.draggingIndex}`);
+            return;
+        }
+
+        // Create a copy of the values array
+        const newValues = [...field.values];
+
+        // Remove the dragging item from its original position
+        newValues.splice(this.draggingIndex, 1);
+
+        // Insert the dragging item into the new position
+        newValues.splice(index, 0, draggedPath);
+
+        this.updateFragment({
+            target: {
+                multiline: true,
+                dataset: { field: currentFieldName },
+            },
+            values: newValues,
+        });
+
+        // Remove dragover class from all elements
+        this.#removeAllDragoverClasses();
+
+        // Reset drag state
+        this.draggingIndex = -1;
+        this.draggingFieldName = null;
+
+        // Request an update to reflect changes
+        this.requestUpdate();
+    }
+
+    // Helper method to remove all dragover classes
+    #removeAllDragoverClasses() {
```

Comment:
are the multiple queries needed?
Why not:
```suggestion
#removeAllDragoverClasses() {
    // Remove from host element
    this.classList.remove('dragover');
    
    // One query handles all elements with 'dragover' class
    if (this.shadowRoot) {
        this.shadowRoot.querySelectorAll('.dragover').forEach(element => 
            element.classList.remove('dragover')
        );
    }
}
```